src/rtkpos.c

```diff
@@ -1593,17 +1600,18 @@ static void holdamb(rtk_t *rtk, const double *xa)
     for (i=0;i<nv;i++) R[i+i*nv]=rtk->opt.varholdamb;
         
     /* update states with constraints */
+    sparse_mat_compress(H);
```

Comment:
The only consumer of the sparse matrix appears to be filter(), and here the sparse matrix is freed after this call, so it appears that the call to sparse_mat_compress() could be avoided here.